.pre-commit-config.yaml

```diff
@@ -78,6 +78,23 @@ repos:
     types: [rust]
     args: [--]
 
+  - id: clippy
+    name: Rust code quality check
+    description: Lint rust sources
+    entry: cargo clippy
+    language: system
+    args: [--workspace, --all-features, --, -D, warnings]
+    types: [rust]
+    pass_filenames: false
+
+  - id: check-for-openssl
+    name: check-for-openssl
+    description: Lint rust sources
+    entry: ./tools/check_for_openssl.sh
+    language: system
+    types: [rust]
+    pass_filenames: false
```

Comment:
I moved these up because these are faster than `cargo check`, so if they indicating any linting issues, we can fix them sooner